### PR TITLE
Use service connector for boto session if possible

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
This PR updates the `AWSContainerRegistry` so it uses its configured service connector to describe the repositories instead of local AWS credentials.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/aws/container_registries/aws_container_registry.py [hunk 1]
```diff
@@ -61,6 +69,35 @@ def _get_region(self) -> str:
...
+                    model=model
+                )
+                assert isinstance(connector, AWSServiceConnector)
+                return connector.get_ecr_client()
+            except Exception as e:
+                logger.error(
+                    "Unable to get ECR client from service connector: %s",
+                    str(e),
+                )
+
```
Comment (stefannica) [inline src/zenml/integrations/aws/container_registries/aws_container_registry.py:95]: return None ?

File: src/zenml/integrations/aws/container_registries/aws_container_registry.py [hunk 2]
```diff
@@ -76,10 +113,9 @@ def prepare_image_push(self, image_name: str) -> None:
             raise ValueError(f"Invalid docker image name '{image_name}'.")
         repo_name = match.group(1)
 
+        client = self._get_ecr_client()
```
Comment (stefannica) [inline src/zenml/integrations/aws/container_registries/aws_container_registry.py:116]: client might be None

**Review by stefannica — CHANGES_REQUESTED** [review]
Everything seems fine, nothing major to add on top of what the linter would also surely reveal as typing errors.

File: src/zenml/integrations/aws/container_registries/aws_container_registry.py (hunk 2 above)
Comment (schustmi) [reply src/zenml/integrations/aws/container_registries/aws_container_registry.py:116]: Currently it can not be None, see explanation above

**Review by stefannica — APPROVED** [review]

File: src/zenml/integrations/aws/container_registries/aws_container_registry.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/integrations/aws/container_registries/aws_container_registry.py:95]: you're right, my mistake

**Pull request merged**